examples/tomo/filtered_backprojection_parallel_2d.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Example using the ray transform with 2d parallel beam geometry."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+import odl
+
+
+# Discrete reconstruction space: discretized functions on the rectangle
+# [-20, 20]^2 with 300 samples per dimension.
+reco_space = odl.uniform_discr(
+    min_corner=[-20, -20], max_corner=[20, 20], nsamples=[200, 200],
+    dtype='float32')
+
+# Make a parallel beam geometry with flat detector
+# Angles: uniformly spaced, n = 360, min = 0, max = 2 * pi
+angle_partition = odl.uniform_partition(0, 2 * np.pi, 360)
+# Detector: uniformly sampled, n = 558, min = -30, max = 30
+detector_partition = odl.uniform_partition(-30, 30, 2000)
+geometry = odl.tomo.Parallel2dGeometry(angle_partition, detector_partition)
+
+# ray transform aka forward projection. We use ASTRA CUDA backend.
+ray_trafo = odl.tomo.RayTransform(reco_space, geometry, impl='astra_cuda')
+# fourier transform in detector direction
+fourier = odl.trafos.FourierTransform(ray_trafo.range, axes=[1])
+
+# Create filter
+theta_max = detector_partition.extent() / 2.0
+
+
+def window(x):
+    # Select cutoff value
+    cutoff = 0.7
+    nonzero = np.abs(x) < theta_max * cutoff
+    return nonzero * (1 + np.cos(np.abs(x) / (theta_max / cutoff)))
+
+
+ramp_function = fourier.range.element(lambda x: 0*x[0] + np.abs(x[1]) * window(x[1]))
```

Comment:
I'm not quite sure how to do this. It does the automatic broadcasting if you "do enough" to reach the required shape. One way I could think of is to check the shape which comes out, compare it to the required shape, and if the output shape is a "subset", we could fill in the zero-addition broadcasting operations in the "gaps".